node/src/main.rs

```diff
@@ -598,133 +746,435 @@ async fn main() -> Result<(), Box<dyn Error>> {
-                                                 warn!(
-                                                     received = %genesis.len(),
-                                                     peer = %peer,
-                                                     "Genesis bead count mismatch"
-                                                 );
-                                             }
-                                         }
-                                     }
-                                     bead::BeadResponse::Error(error) => {
-                                         error!(error = ?error, "Bead sync response error");
-                                         peer_manager.update_score(&peer, -1.0);
-                                     }
-                                 };
-                             }
-                         }
-                     }
+                    request_response::Event::Message {
+                        peer,
+                        message,
+                        connection_id,
+                    },
+                )) => {
+                    info!(
+                        peer = %peer,
+                        message = ?message,
+                        connection = ?connection_id,
+                        "Bead sync message received"
+                    );
+                    match message {
+                        request_response::Message::Request {
+                            request,
+                            request_id: _,
+                            channel,
+                        } => {
+                            // Handle the bead sync request here
+                            match request {
+                                bead::BeadRequest::GetBeads(hashes) => {
+                                    let mut beads = Vec::new();
+                                    {
+                                        let braid_lock = braid.read().await;
+                                        for hash in hashes.iter() {
+                                            if let Some(index) =
+                                                braid_lock.bead_index_mapping.get(hash)
+                                            {
+                                                if let Some(bead) = braid_lock.beads.get(*index) {
+                                                    beads.push(bead.clone());
+                                                }
+                                            }
+                                        }
+                                    }
+                                    //Sending all the beads requested in the hashes supplied during `GetData` request
+                                    swarm.behaviour_mut().respond_with_beads(channel, beads);
+                                }
+                                bead::BeadRequest::GetTips => {
```

Comment:
We need to check the `ibd_spinlock` here and refuse to respond with tips if we're in IBD. Add an error case `BeadSyncError::PeerSyncing` for this.